lib/celluloid/proxy/actor.rb

```diff
@@ -19,6 +19,14 @@ def inspect
     "#<Celluloid::Proxy::Actor(#{@mailbox.address}) dead>"
   end
 
+  def eql? other
+    @mailbox.eql? other.mailbox
```

Comment:
I'd suggest checking `class` too, otherwise all subclasses of proxy will compare equal...